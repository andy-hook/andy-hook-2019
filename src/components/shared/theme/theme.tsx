import React, { memo, ReactNode } from "react"
import { ThemeProvider } from "styled-components"
import { IStore } from "store"
import { connect } from "react-redux"
import { themes } from "@style/theme"
import { TThemeType, ITheme } from "@custom-types/theme"

interface Props {
  children: ReactNode
  themeType: TThemeType
}

type AllProps = Props & Partial<IStore>

const mapStateToProps = ({ primaryTheme, secondaryTheme }: IStore) => {
  return { primaryTheme, secondaryTheme }
}

const Theme: React.FunctionComponent<AllProps> = memo(
  ({ children, themeType, primaryTheme = "dark", secondaryTheme = "dark" }) => {
    const getTheme = (type: string): ITheme => {
      if (type === "primary-theme") {
        return themes[primaryTheme]
      } else if (type === "secondary-theme") {
        return themes[secondaryTheme]
      } else {
        return themes[primaryTheme]
      }
    }

    return (
      <ThemeProvider theme={getTheme(themeType)}>
        <>{children}</>
      </ThemeProvider>
    )
  }
)

const ConnectedTheme = connect(mapStateToProps)(Theme)
export default ConnectedTheme
