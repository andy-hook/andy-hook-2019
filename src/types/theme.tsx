export type TGrey = 100 | 200 | 300 | 400 | 500 | 600 | 700 | 800

export interface IGrey {
  [key: number]: string
  100: string
  200: string
  300: string
  400: string
  500: string
  600: string
  700: string
  800: string
}

export interface ITheme {
  name: "light" | "dark"
  text: IGrey
  tone: IGrey
}
