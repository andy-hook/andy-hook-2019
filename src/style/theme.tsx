import { ITheme, TGrey } from "@custom-types/theme"
import { css } from "styled-components"

import { lightGreyHSL, darkGreyHSL } from "@style/variables"

export const lightTheme: ITheme = {
  name: "light",
  text: darkGreyHSL,
  tone: lightGreyHSL,
}

export const darkTheme: ITheme = {
  name: "dark",
  text: lightGreyHSL,
  tone: darkGreyHSL,
}

export const themeTone = (value: TGrey) => css`
  ${props => `hsl(${props.theme.tone[value]})`}
`
export const themeToneAlpha = (value: TGrey, alpha: number) => css`
  ${props => `hsla(${props.theme.tone[value]}, ${alpha})`}
`
export const themeText = (value: TGrey) => css`
  ${props => `hsl(${props.theme.text[value]})`}
`
export const themeTextAlpha = (value: TGrey, alpha: number) => css`
  ${props => `hsla(${props.theme.text[value]}, ${alpha})`}
`

export const themeName = css`
  ${props => props.theme.name}
`

export const isDarkTheme = (output: string) => css`
  ${props => props.theme.name === "dark" && output}
`
