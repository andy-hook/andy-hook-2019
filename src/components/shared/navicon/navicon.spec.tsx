import React from "react"
import renderer from "react-test-renderer"
import { ThemeProvider } from "styled-components"
import { darkTheme } from "@style/theme"

import Navicon from "./navicon"

describe("<Navicon />", () => {
  it("renders correctly", () => {
    const tree = renderer
      .create(
        <ThemeProvider theme={darkTheme}>
          <Navicon onClick={jest.fn()} />
        </ThemeProvider>
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
