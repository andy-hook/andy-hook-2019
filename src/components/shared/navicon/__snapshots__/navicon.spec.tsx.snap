// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Navicon /> renders correctly 1`] = `
<button
  className="sc-bdVaJa hkySjj"
  onClick={[MockFunction]}
/>
`;
